Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -345,6 +347,13 @@ class Gx2Fitter {
     /// Calibration context for the fit
     const CalibrationContext* calibrationContext{nullptr};
 
+    /// The current iteration of the fitter
+    size_t nUpdate = Acts::MultiTrajectoryTraits::kInvalid;
```

Comment:
Using the `kInvalid` value here as a sentinel can be a bit dangerous if you're not careful. I don't see where this is modified, is the sentinel handled safely? Naively I'd have thought you'd want to start at zero?

Reply:
I put `kInvalid` to not leave it uninitialised. The variable is used to forward the current iteration. In case we are in the first iteration we need to add trackstates at each surface.

I will also add a more detailed comment.